packages/frontend/tests/CreateStore.test.tsx

```diff
@@ -0,0 +1,145 @@
+
+      const nameInput = screen.getByTestId(`storeName`);
+      const descInput = screen.getByTestId(`desc`);
+      const chains = screen.getByRole("button", { name: /hardhat/i });
+      const uploadInput = screen.getByTestId("file-upload");
+      const saveBtn = screen.getByRole("button", { name: /save/i });
+
+      await user.upload(uploadInput, file);
+
+      await user.click(chains);
+      await user.clear(nameInput);
+      await user.clear(descInput);
+      await user.type(nameInput, "New Store Name!!");
+      await user.type(descInput, "New Store Description!!");
+      await user.click(saveBtn);
+    });
+    await waitFor(async () => {
+      expect(spy).toHaveBeenCalled();
+      let count = 0;
+      let pId;
+      for await (const [id, tag] of sm.tags.iterator()) {
+        count++;
+        //Creating store thru create-store page should auto generate published and removed tags
+        if (tag.name === "visible") {
+          pId = id;
+        } else {
+          expect(tag.name).toEqual("remove");
+        }
+      }
+      expect(count).toEqual(2);
+      const manifest = await sm.manifest.get();
+      expect(manifest.publishedTagId).toEqual(pId);
+      expect(manifest.name).toEqual("New Store Name!!");
+      expect(manifest.description).toEqual("New Store Description!!");
+    });
+  });
+
+  test("create store - with changed payee address and base currency", async () => {
+    render(<CreateStore />, sm);
+    expect(spy).not.toHaveBeenCalled();
+    const payee = randomAddress();
+    await waitFor(async () => {
+      const file = new File(["hello"], "hello.png", { type: "image/png" });
+
+      const nameInput = screen.getByTestId(`storeName`);
+      const descInput = screen.getByTestId(`desc`);
+      const chains = screen.getByRole("button", { name: /hardhat/i });
+      const uploadInput = screen.getByTestId("file-upload");
+      const saveBtn = screen.getByRole("button", { name: /save/i });
+      const payeeInput = screen.getByTestId("payeeAddress");
+      const baseAddrInput = screen.getByTestId("baseTokenAddr");
+
+      await user.upload(uploadInput, file);
+      await user.click(chains);
+      await user.clear(nameInput);
+      await user.clear(descInput);
+      await user.clear(payeeInput);
+      await user.clear(baseAddrInput);
+
+      await user.type(nameInput, "New Store Name II");
+      await user.type(descInput, "New Store Description II");
+      await user.type(payeeInput, payee);
+      await user.type(baseAddrInput, zeroAddress);
+
+      await user.click(saveBtn);
+    });
+    await waitFor(async () => {
+      expect(spy).toHaveBeenCalled();
+      const manifest = await sm.manifest.get();
+      expect(manifest.name).toEqual("New Store Name II");
+      expect(manifest.description).toEqual("New Store Description II");
+      expect(manifest.payee[0].addr).toEqual(payee);
+      expect(manifest.setBaseCurrency?.tokenAddr).toEqual(zeroAddress);
+    });
```

Comment:
i'm wondering if this is enough of check to see if we created a store but it might be.

just asking to understand whether the other steps to the process (mint store nft, upload metadata) are tested elsewhere.

Reply:
@cryptix this actually tests the minting! Because it has to mint the store and pass permission check in order to enroll the keycard, and then once kc is enrolled successfully it will create/update the shop manifest because of the conditionals in the create-store file